src/mimefactory.rs

```diff
@@ -1618,4 +1618,57 @@ mod tests {
             .await
             .unwrap();
     }
+
+    #[test]
+    fn test_no_empty_lines_in_header() {
+        // See https://github.com/deltachat/deltachat-core-rust/issues/2118
+        let to_tuples = vec![
+            ("Nnnn", "nnn@ttttttttt.de"),
+            ("😀 ttttttt", "ttttttt@rrrrrr.net"),
+            ("dididididididi", "t@iiiiiii.org"),
+            ("Ttttttt", "oooooooooo@abcd.de"),
+            ("Mmmmm", "mmmmm@rrrrrr.net"),
+            ("Zzzzzz", "rrrrrrrrrrrrr@ttttttttt.net"),
+            ("Xyz", "qqqqqqqqqq@rrrrrr.net"),
+            ("", "geug@ttttttttt.de"),
+            ("qqqqqq", "q@iiiiiii.org"),
+            ("bbbb", "bbbb@iiiiiii.org"),
+            ("", "fsfs@iiiiiii.org"),
+            ("rqrqrqrqr", "rqrqr@iiiiiii.org"),
+            ("tttttttt", "tttttttt@iiiiiii.org"),
+            ("", "tttttt@rrrrrr.net"),
+        ];
+        let mut to = Vec::new();
+        for (name, addr) in to_tuples {
```

Comment:
This can be written as `to = to_tuples.map(|(name, addr)| if name.is_empty() { Address::new_mailbox(addr.to_string()) } else { Address::new_mailbox_with_name(name.to_string(), addr.to_string()) }).collect()`